### ISSUE TITLE
[go_router] GoRouter-wide pageBuilder 

### ISSUE THREAD

**Issue by chunhtai** [flutter/flutter#99120]
copied from https://github.com/csells/go_router/issues/204

Would be useful for things like different transitions per target platform (even if it's all material), making everything fade, etc. Since everything uses a core pageBuilder anyway, this wouldn't be too hard to do...

**Comment by mateusfccp** [comment]
Yeah, it makes no sense to only be able to change transitions on a per-route base.

This is something too much fundamental...
